src/main/java/io/strimzi/ValidatingWebhook.java

```diff
@@ -23,15 +23,13 @@
 import javax.ws.rs.core.MediaType;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.regex.Pattern;
 
 @Path("/drainer")
 public class ValidatingWebhook {
     private static final Logger LOG = LoggerFactory.getLogger(ValidatingWebhook.class);
 
-    private static final Pattern ZOOKEEPER_PATTERN = Pattern.compile(".+(-zookeeper-)[0-9]+");
-    private static final Pattern KAFKA_PATTERN = Pattern.compile(".+(-kafka-)[0-9]+");
-
+    private static final String KIND_LABEL_VALUE = "Kafka";
+    private static final String KIND_LABEL_KEY = "strimzi.io/kind";
```

Comment:
I don't think the `strimzi.io/kind` label works. That would capture also all the Deployment managed pods. You need to target the `strimzi.io/name` label I think and match it against something like `.+(-zookeeper)` and `.+(-kafka)`.

Reply:
> I don't think the strimzi.io/kind label works. 

Correct, that label also includes the entity operator pod for example.

Instead, the value of `strimzi.io/name` is created with the pattern $CLUSTER_NAME-kafka or $CLUSTER_NAME-zookeeper. While $CLUSTER_NAME part is variable, the second part (-kafka and -zookeeper) is constant and that's what you can use for matching.